### PR TITLE
Add answer's sources

### PR DESCRIPTION
**Pull request by Errorname**
Preview:

image

image

### CODE REVIEW

File: server/src/answer/flags.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  return new Promise((resolve, reject) => {
+    graphcool
+      .request(getFlagsQuery, { answerId: answer.answerId })
+      .then(data => {
+        resolve(data.Answer.node.flags)
+      })
+  })
+}
+
+const deleteFlags = (graphcool, answer) => {
```
Comment (frinyvonnick) [inline server/src/answer/flags.js]: Suggestion:

With async/await you could write more concise and readable code. What do you think about it ?

```js
const deleteFlags = async (graphcool, answer) => {
  const flags = await getFlags(graphcool, answer)
  await Promise.all(flags.map(({ id: flagId }) => {
    return graphcool.request(deleteFlagQuery, { flagId })
  }))
}
```

File: src/scenes/Question/scenes/Answer/components/Source/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <i
+              className="material-icons"
+              style={{ marginLeft: '6px', paddingTop: '6px' }}
+            >
+              close
+            </i>
+          }
+          style={{ minWidth: 'initial', alignItems: 'center' }}
+          accent
+          onClick={() => props.delete()}
```
Comment (frinyvonnick) [inline src/scenes/Question/scenes/Answer/components/Source/index.js]: Why don't you pass `delete` directly ? `onClick={props.delete}`

File: src/scenes/Question/scenes/Answer/components/Sources/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+class Sources extends Component {
+  addSource () {
+    const { sources, handleChange } = this.props
+    handleChange(sources.concat([{ id: uuid(), label: '', url: '' }]))
+  }
+
+  changeLabel (id, label) {
+    const { sources, handleChange } = this.props
+    handleChange(
```
Comment (frinyvonnick) [inline src/scenes/Question/scenes/Answer/components/Sources/index.js]: you could currify handleChange to make it reusable:

```js
const changeSomething = (id, changes) => sources => {
  handleChange(
    sources.map(s => {
      if (s.id !== id) return s
      return { ...s, ...changes }
    })
  )
}
```

```js
changeLabel (id, label) {
  const { sources, handleChange } = this.props
  changeSomething(id, { label })(sources)
}
```

What do you think ?

File: server/src/answer/flags.js (hunk 1 above)
Comment (Errorname) [reply server/src/answer/flags.js]: Yes, you are right. I still have to wrap my head around async/await !

File: src/scenes/Question/scenes/Answer/components/Source/index.js (hunk 2 above)
Comment (Errorname) [reply src/scenes/Question/scenes/Answer/components/Source/index.js]: Simple oversight after a refact, will fix!

File: src/scenes/Question/scenes/Answer/components/Sources/index.js (hunk 3 above)
Comment (Errorname) [reply src/scenes/Question/scenes/Answer/components/Sources/index.js]: I like that !

**Comment by hgwood** [comment]
I like the feature, though I think it might be taking too much space on the UI. The important part is the answer. Maybe you could tweak the font sizes. This can be done in further PRs, don't block the feature for this :).

**Comment by Errorname** [comment]
I agree with you @hgwood. Up until now, I didn't have a clear UI/UX mindset for this project.

However, yesterday, @bpetetot and I took the time to think about it and I now have a long list of changes to do in order to have a coherent interface.

As I will remove react-toolbox and break things during the implementation of the new interface, this will be the subject of an other PR (and some demos to get feedback too)

Current roadmap:
* Merge sources (soon)
* Finish email implementation (in the next days)
* Rework interface (probably next week)

File: src/scenes/Question/scenes/Answer/components/Sources/index.js (hunk 3 above)
Comment (Errorname) [reply src/scenes/Question/scenes/Answer/components/Sources/index.js]: In the end, I don't think I need a currifying method, I can just do this:
```js
  changeAttribute = (id, attributes) => {
    const { sources, handleChange } = this.props
    handleChange(
      sources.map(s => {
        if (s.id !== id) return s
        return { ...s, ...attributes }
      })
    )
  }

  changeLabel(id, label) {
    this.changeAttribute(id, { label })
  }
```

Comment (frinyvonnick) [reply src/scenes/Question/scenes/Answer/components/Sources/index.js]: 👍

**Pull request merged**